### ISSUE TITLE
ATL transforms are drawn with nearest-neighbor filtering

### ISSUE THREAD

**Issue by eevee** [renpy/renpy#1403]
Script:

```
image white = '#fff'
image scrap = "scrap.png"

transform centerleft:
    xcenter 0.25
    ycenter 0.5

transform centerright:
    xcenter 0.75
    ycenter 0.5

transform zoomby(n):
    zoom n

label start:
    scene white
    show scrap as scrap_orig at centerleft
    show scrap as scrap_zoom at truecenter, zoomby(0.25)
    show expression im.FactorScale(Image("scrap.png"), 0.25) as scrap_scale at centerright
    "Hmm..."
```

Result:

![Screenshot of the original image, plus two versions of it scaled down 4×; the first has awkward jagged edges, but the second looks smooth](https://user-images.githubusercontent.com/94112/36953835-7d1c5774-1fd2-11e8-9bbd-6d78497ddc55.png)

The ATL version looks pretty similar to what I get if I scale the image down by hand in GIMP.  I've also noticed that a rotated rectangle (e.g. a Solid) doesn't get anti-aliasing along its edges.

`renpy.config.nearest_neighbor` is at its default of `False`, and `renpy.display.draw` is a `renpy.gl.gldraw.GLDraw`.

**Comment by renpytom** [comment]
Actually, this is bilinear scaling, it's just that bilinear scaling has problems when you use it to scale down below 50%.

Bilinear sampling, in a GPU, samples from up to four pixels. But when you go down to 0.25% of the normal size, you're sampling from  more than that - it would have to sample from 16 pixels to create one, which would be slow. 

The right solution would be to mipmap the image, but Ren'Py doesn't support that yet. 

im.FactorScale gets this right, but it has a lot more time to do its work. If you want Transform to scale in realtime, you need to limit yourself to a zoom of .5 or more.

**Comment by eevee** [comment]
Ah, that makes sense.

Comparing the output of both at 50%, I notice that `FactorScale` appears to be blending with the colors of transparent pixels (in this case black), which makes the outline slightly grayer than it ought to be...  but the responsible code repeatedly insists it has no knowledge of which channel is actually alpha, so maybe that's not readily fixable.

**Comment by renpytom** [comment]
This has been fixed in 7.4 with the model-based renderer.
